src/lib.rs

```diff
@@ -267,9 +266,10 @@ extern "C" fn init(callbacks: *mut PluginCallbacks, config_path: *const c_char)
                 message_threads
             };
 
+            let mut senders = Vec::with_capacity(num_threads);
             for i in 0..num_threads {
                 let (messages_tx, messages_rx) = mpsc::sync_channel(0);
-                senders.push(messages_tx.clone());
+                senders.push(messages_tx);
```

Comment:
`messages_tx` is added to senders Vec, then the Vec is set on the static variable.
There is no need to clone `messages_tx`, this is not moved to the created thread. Only `messages_rx` is moved to the created thread.